src/main/java/com/internetitem/logback/elasticsearch/writer/ElasticsearchWriter.java

```diff
@@ -42,17 +47,41 @@ public void write(char[] cbuf, int off, int len) {
 		sendBuffer.append(cbuf, off, len);
 
 		if (sendBuffer.length() >= settings.getMaxQueueSize()) {
-			errorReporter.logWarning("Send queue maximum size exceeded - log messages will be lost until the buffer is cleared");
+			errorReporter.logWarning(
+					"Send queue maximum size exceeded - log messages will be lost until the buffer is cleared");
 			bufferExceeded = true;
 		}
 	}
 
-	public void sendData() throws IOException {
+	public void sendData() throws IOException, NoSuchAlgorithmException, KeyManagementException {
 		if (sendBuffer.length() <= 0) {
 			return;
 		}
 
-		HttpURLConnection urlConnection = (HttpURLConnection)(settings.getUrl().openConnection());
+		HttpURLConnection urlConnection;
+		if ("https".equalsIgnoreCase(settings.getUrl().getProtocol())) {
+			if (!settings.isValidateCert()) {
+				//accept all certs
+				TrustManager[] trustAllCerts = new TrustManager[] { new X509TrustManager() {
+					public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+						return null;
+					}
+
+					public void checkClientTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+					}
+
+					public void checkServerTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+					}
+				} };				
+				SSLContext sc = SSLContext.getInstance("SSL");
+			    sc.init(null, trustAllCerts, new java.security.SecureRandom());
+			    HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());				
```

Comment:
This could potentially disturb other parts of an application that weren't expecting all `HttpsURLConnection`s (in addition to ones used by Logback) to suddenly disable all certificate validation.